mergebot.py

```diff
@@ -46,33 +125,49 @@ def main():
                 l.info('{} Successfully Read'.format(filename))
                 configs.append(config)
             except yaml.YAMLError as exc:
-                l.error('Error parsing file {}: {}.'.format(filename, exc))
-                l.error('Please fix and try again.')
-                return
-    # Workaround for multiprocessing SIGINT problems per
-    # http://stackoverflow.com/questions/11312525 and the like. Children need to
-    # ignore SIGINT; parent should obey and clean up itself.
-    original_sigint_handler = signal.signal(signal.SIGINT, signal.SIG_IGN)
-    pool = Pool(len(configs))
-    signal.signal(signal.SIGINT, original_sigint_handler)
+                l.fatal(
+                    'Error parsing file {filename}: {exc}. Please fix and try '
+                    'again.'.format(filename=filename, exc=exc))
+    return configs
 
-    try:
-        l.info('Starting Poller Pool')
-        pool.map_async(poll_scm, configs)
-        while True:
-            input('Mergebot running; press Ctrl + C to cancel.\n')
-    except KeyboardInterrupt:
-        l.info('Exiting.')
-        pool.terminate()
-    else:
-        # Generally this shouldn't happen - pollers should run forever.
-        pool.close()
-        pool.join()
-        # TODO(jasonkuster): We could get into a state where we're only polling
-        # one repo and all others have crashed. Once mergebot is functional,
-        # work on better poller management.
-        l.error('Mergebot terminated early. All pollers must have crashed.')
 
+class PollerInfo(object):
+    """MergerInfo contains important hooks for the pollers we are running."""
+    def __init__(self, process, pipe):
+        self.process = process
+        self.pipe = pipe
+
+
+def poll_scm(config, pipe):
+    """poll_scm handles delegating a single repository's work to an SCM poller.
+
+    Args:
+        config: A dictionary of configuration to use for the poller.
+        pipe: Communication pipe for passing messages.
+    """
+    poller = mergebot_poller.create_poller(config, pipe)
+    poller.poll()
+
+
+def start_pollers(configs):
+    """start_pollers starts a set of pollers for specified configurations.
+    
+    Args:
+        configs: Configurations for the pollers.
+
+    Returns:
+        Array of poller info (process, comm pipe).
+    """
+    pollers = []
+    for config in configs:
+        parent_pipe, child_pipe = Pipe()
+        p = Process(target=poll_scm, args=(config, child_pipe,))
```

Comment:
I will look into ways to make this work. One way I considered was to have a DBPublisher class, whose constructor would create a new handle to the database. Does that sound like a good solution here?